lib/player.js

```diff
@@ -1949,10 +1949,12 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
         // received.
         const abrManagerFactory = preloadManager.getAbrManagerFactory();
         if (abrManagerFactory) {
-          if (!this.abrManagerFactory_ ||
+          if (this.abrManagerFactory_ &&
               this.abrManagerFactory_ != abrManagerFactory) {
-            this.abrManager_ = preloadManager.receiveAbrManager();
-            this.abrManagerFactory_ = preloadManager.getAbrManagerFactory();
+            this.abrManager_.stop();
```

Comment:
2 things:
1. ABR manager can be null here (or even it always is?)
2. If we're getting rid of this instance we should call `release()`, not `stop()`.

Reply:
1. I added a check
2. It can still be reused in another preload, so it's better to use stop.